GPG encryption: encrypted AIPs identified

jrwdunham opened the pull request:
AIPs in GPG-encrypted spaces are marked as "encrypted" in the archival storage tab.

See https://github.com/artefactual/archivematica-storage-service/pull/192

sevein commented on src/dashboard/src/templates/archival_storage/list.html:
```diff
@@ -96,7 +96,11 @@
         {% for item in aips %}
           <tr>
-            <td><a href="{% url 'view_aip' item.uuid %}">{{ item.name }}</a></td>
+              {% if item.encrypted %}
```
Our indentation in HTML files uses two spaces instead of four.

sevein commented on src/dashboard/src/templates/archival_storage/list.html:
```diff
@@ -96,7 +96,11 @@
           <tr>
-            <td><a href="{% url 'view_aip' item.uuid %}">{{ item.name }}</a></td>
+              {% if item.encrypted %}
+              <td><a href="{% url 'view_aip' item.uuid %}">{{ item.name }} ({% trans "encrypted" %})</a></td>
```
I think it's better to do as follows:

```django
{% blocktrans with aip_name=item.name %}{{ aip_name }} encrypted{% endblocktrans %}
```

With that the translator sees `{{ aip_name }} encrypted` and they can freely move the placeholder and fully control the structure of the sentence.

jrwdunham replied:
Ok. I see you removed the parens too. Did you mean to do that? To me, they indicate that the "encrypted" string is a qualifier on the name of the AIP and not the part of the name itself, i.e., "My AIP (encrypted)" vs "My Other AIP". Are you fine with::

```
    {% blocktrans with aip_name=item.name %}{{ aip_name }} (encrypted){% endblocktrans %}
```

and then the translator can remove the parens if they see fit?

sevein replied:
Oh I didn't see the parens! Sorry about that! `{{ aip_name }} (encrypted)` looks better to me but I think both ways work, up to you.

jrwdunham replied:
Ok. I think I've done that. See 4e74f713e88b1863e6db745ae7b6f20ca3f48e52

sevein commented:
Rebased.

jraddaoui commented on src/archivematicaCommon/lib/storageService.py:
```diff
@@ -238,6 +238,36 @@ def get_files_from_backlog(files):
+        response = _storage_api_session().get(url, params=params)
+        spaces = response.json()
+        return_spaces += spaces['objects']
+        if not spaces['meta']['next']:
```
We're trusting in a response field to end this `while` loop, which someday may end up being an endless loop. Do we have a way to get the total count of spaces from the SS?

jrwdunham replied:
I take your point, but we're trusting that the tastypie API to always return a boolean for `['meta']['next']` and to eventually return `False`. I think this is ok. The weaker argument is that this is how this was done previously (this is old code) and how it still is done in other parts of this module. It's moot now though because I'm removing this code in favour of that which indicates encryption status using ES AIP indexing.

jraddaoui commented on src/dashboard/src/components/archival_storage/views.py:
```diff
@@ -566,6 +568,38 @@ def es_pager(page, page_size):
+        if not space:
+            continue
+        if space['access_protocol'] == 'GPG':
+            aip['encrypted'] = True
```
Could we return this property directly from the SS?

jrwdunham replied:
Yes, and this is essentially what sevein advised too. I'll modify this PR to indicate AIP encryption status in the Elasticsearch index. This way it is computed once and will require no extra requests.

jraddaoui reviewed (commented):
Looks good to me @jrwdunham!

But I think it's a little overkill just to know which AIPs are encrypted. Could we add 'encrypted' as a property to the AIPs in the SS, so it's returned directly while fetching them?

jrwdunham committed:
Encrypted AIPs ES-indexed and GUI-displayed

When an AIP is indexed, its encrypted status (boolean) is included. This
information is then displayed in the archival storage tab in list
browse, search results and detail view.

jraddaoui reviewed (approved):
Great! Thank you @jrwdunham!
